src/main/webapp/search.js

```diff
@@ -43,31 +56,95 @@ async function getSearchResultsHelper(document, window) {
     }
 
     if(topic != null) {
-        await fetch("/search?topic="+topic).then(response => response.json()).then((results) => {
-            var resultContainer = document.getElementById("result-container");
+        var tutors = getTutors(topic);
+        var books = getBooks(topic);
+        
+        await tutors;
+        await books;
```

Comment:
What are these for?

Reply:
They just make sure to wait for the functions, getTutors and getBooks, to return before redirectToResult returns.